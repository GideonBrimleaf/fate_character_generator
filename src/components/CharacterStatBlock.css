.stat-block {
  list-style-type: none;
  padding-left: 0px;
}

.character-sheet-header {
  margin-bottom:0rem;
}

.stat-table {
  margin: 0px auto;
}

.align-right {
  text-align: right;
}

.align-left {
  text-align: left;
}

.fred, .fred td, .fred tr {
  border: 1px solid red;
}