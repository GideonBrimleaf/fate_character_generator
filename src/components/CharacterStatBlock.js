import React from 'react'
import './CharacterStatBlock.css'

const CharacterStatBlock = (props) => {

  const statItemNames = Object.keys(props.characterStats)

  const prettifyName = (objectKey) => {
    const capitalisedFirstLetter = objectKey.charAt(0).toUpperCase() + objectKey.slice(1)
    const splitName = capitalisedFirstLetter.split(/(?=[A-Z])/)
    return splitName.join(' ')
  }

  const statItems = statItemNames.map(stat => {
    return (
      <tr key={prettifyName(stat)}>
        <td className='align-right'>{ prettifyName(stat) }:</td>
        <td className='align-left'>{ props.characterStats[stat] }</td>
      </tr>
    )
  })

  return (
    <>
      <table className='fred'>
        <caption>{ props.characterStatCategory }</caption>
        <tbody>
          { statItems }
        </tbody>
      </table>
    </>
  )
}

export default CharacterStatBlock