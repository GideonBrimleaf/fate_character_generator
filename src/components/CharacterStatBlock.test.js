/* eslint-disable react/jsx-props-no-spreading */

import React from 'react';
import { render } from '@testing-library/react';

import CharacterStatBlock from './CharacterStatBlock';

it('renders correctly', () => {
  // Arrange
  const props = {
    characterStats: {
      expectedName: 'Expected value',
    },
    statBlock: 'expected-block',
  };

  // Act
  const result = render(
    <CharacterStatBlock {...props} />,
  );

  // Assert
  expect(result.asFragment()).toMatchSnapshot();
});
