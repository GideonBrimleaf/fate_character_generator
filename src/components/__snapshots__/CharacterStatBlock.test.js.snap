// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <table
    class="padded-table"
  >
    <caption
      class="table-header"
    />
    <tbody>
      <tr>
        <td
          class="stat-name"
        >
          Expected Name:
        </td>
        <td
          class="expected-block-stat-description"
        >
          Expected value
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
